### DISCUSSION TITLE
How to set custom PartProperties?

### DISCUSSION THREAD

**Discussion by beth-soptim** [phax/phase4#172] [Q&A] [answered]
We set `BDEWPayloadParams` via `payload(AS4OutgoingAttachment.Builder, BDEWPayloadParams)` on `Phase4BDEWSender.AbstractBDEWUserMessageBuilder`. But it seems that the required Part Properties are not contained in the outgoing generated XML message.

`createMainAttachment` in `Phase4BDEWSender.AbstractBDEWUserMessageBuilder` is neither called from Phase4 nor our application. This is the method in `Phase4BDEWSender.AbstractBDEWUserMessageBuilder` that writes the Part Properties.

Is it required to set the payload (via `AS4OutgoingAttachment.Builder`) and the `BDEWPayloadParams` differently?

Thanks in advance.

**Comment by phax** [comment, marked as answer]
What a stupid mistake - mea culpa. Will be part of the 2.4.0 release

Fixed in https://github.com/phax/phase4/commit/004e026860e70f574a7462fbd66302fa8081e72f

**Comment by beth-soptim** [comment]
Hi,

thanks. That was very fast :)

**Reply by phax** [reply]
Please note: the 2.3.1-SNAPSHOT is in reality a 2.4.0-SNAPSHOT and contains a backward incompatible API change (see [News and Noteworthy ](https://github.com/phax/phase4/wiki/News-and-noteworthy))
